src/core/Unif.ml

```diff
@@ -500,7 +500,7 @@ module Inner = struct
     (*Format.printf "(@[unif_rec@ :t1 `%a`@ :t2 `%a`@ :op %a@ :subst @[%a@]@ :bvars %a@])@."
       (Scoped.pp T.pp) (t1,sc1) (Scoped.pp T.pp) (t2,sc2)
       pp_op op US.pp subst B_vars.pp bvars;*)
-    assert (not (T.is_a_type t1 && Type.is_forall (Type.of_term_unsafe t1)));
+    (* assert (not (T.is_a_type t1 && Type.is_forall (Type.of_term_unsafe t1))); *)
```

Comment:
I'm still uneasy with this. Binding variables to stuff that has a quantified type? :s

Reply:
Ok, I will try to fix this first. One place that calls unification with these kinds of terms is the subsumption code (Literal.ml, line 418). I will fix it there and try to find out whether this is the only place.